rust/tool/src/esp.rs

```diff
@@ -94,31 +125,39 @@ impl EspGenerationPaths {
     }
 }
 
-fn nixos_path(path: impl AsRef<Path>, name: &str) -> Result<PathBuf> {
-    let resolved = path
-        .as_ref()
-        .read_link()
-        .unwrap_or_else(|_| path.as_ref().into());
-
-    let parent_final_component = resolved
-        .parent()
-        .and_then(|x| x.file_name())
-        .and_then(|x| x.to_str())
-        .with_context(|| format!("Failed to extract final component from: {:?}", resolved))?;
-
-    let nixos_filename = format!("{}-{}.efi", parent_final_component, name);
+fn hash<'key, T: Serialize>(inputs: impl IntoIterator<Item = (&'key str, T)>) -> Result<String> {
+    let inputs = serde_json::to_string(&BTreeMap::from_iter(inputs))
+        .context("Failed to serialize inputs")?;
```

Comment:
I don't really like to hash things that way, because it does not properly separate the inputs. And we already depend on `serde_json` anyway.

Reply:
What do you mean by "separate the inputs"? It results in a non-reversible hash anyways, right?

Reply:
What I mean is that I don't particularly like just concatenating different data before hashing, since that can in some cases lead to collisions without actually breaking the hash function. It's probably not applicable here, but as said, we already depend on `serde_json` anyway for reading the bootspec document, and I prefer to just not have to worry about this kind of stuff.